website/docs/getting_started.html.markdown

```diff
@@ -0,0 +1,229 @@
+In order to make requests against the GCP API, you need to authenticate to prove
+that it's you making the request. The preferred method of provisioning resources
+with Terraform is to use a [GCP service account](https://cloud.google.com/docs/authentication/getting-started),
+a "robot account" that can be granted a limited set of IAM permissions.
+
+From [the service account key page in the Cloud Console](https://pantheon.corp.google.com/apis/credentials/serviceaccountkey)
+choose an existing account, or create a new one. Next, download the JSON key
+file. Name it something you can remember, and store it somewhere secure on your
+machine.
+
+You supply the key to Terraform using the environment variable
+`GOOGLE_CLOUD_KEYFILE_JSON`, setting the value to the location of the file.
+
+```bash
+export GOOGLE_CLOUD_KEYFILE_JSON={{path}}
+```
+
+-> Remember to add this line to a startup file such as `bash_profile` or
+`bashrc` to store your credentials across sessions!
+
+## Provisioning your resources
+By now, your config will look something like:
+
+```hcl
+provider "google" {
+  project = "{{YOUR GCP PROJECT}}"
+  region  = "us-central1"
+  zone    = "us-central1-c"
+}
+
+resource "google_compute_instance" "vm_instance" {
+  name         = "terraform-instance"
+  machine_type = "f1-micro"
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-9"
+    }
+  }
+
+  network_interface {
+    # A default network is created for all GCP projects
+    network       = "${google_compute_network.vpc_network.self_link}"
+    access_config = {
+    }
+  }
+}
+
+resource "google_compute_network" "vpc_network" {
+  name                    = "terraform-network"
+  auto_create_subnetworks = "true"
+}
+```
+
+With a Terraform config and with your credentials configured, it's time to
+provision your resources:
+
+```hcl
+terraform apply
+```
+
+Congratulations! You've gotten started using the Google provider and provisioned
+a virtual machine on Google Cloud Platform. The key concepts unique to GCP are:
+
+* How a `project` contains resources
+    * and how to use a default `project` in your provider
+* What a resource being global, regional, or zonal means on GCP
+    * and how to specify a default `region` and `zone`
+* How GCP uses `name` and `self_link` to identify resources
+* How to add GCP service account credentials to Terraform
+
+Run `terraform destroy` to tear down your resources.
+
+Afterwards, check out the [provider reference](/docs/providers/google/provider_reference.html) for more details on configuring
+the provider block (including how you can eliminate it entirely!).
+
+You can also check out the [GCP Community tutorials](https://cloud.google.com/community/tutorials/)
+such as:
+* [Managing GCP Projects with Terraform](https://cloud.google.com/community/tutorials/managing-gcp-projects-with-terraform)
+* [Modular Load Balancing with Terraform](https://cloud.google.com/community/tutorials/modular-load-balancing-with-terraform)
```

Comment:
woo published: https://cloud.google.com/community/tutorials/getting-started-on-gcp-with-terraform
want to include in this pr?